packages/react-vapor/src/components/table-hoc/TableWithActions.tsx

```diff
@@ -1,20 +1,56 @@
 import * as React from 'react';
+import {connect} from 'react-redux';
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {TableHOCRowActions} from './actions/TableHOCRowActions';
 import {ITableHOCOwnProps} from './TableHOC';
+import {TableSelectors} from './TableSelectors';
 
 export interface ITableWithActionsProps extends ITableHOCOwnProps {}
 
 type TableWithActionsComponent = React.ComponentClass<ITableWithActionsProps>;
 
 export const tableWithActions = () => (Component: TableWithActionsComponent): TableWithActionsComponent => {
-    class TableWithActions extends React.Component<ITableWithActionsProps> {
+    const mapStateToProps = (state: IReactVaporState, ownProps: ITableHOCOwnProps) => ({
+        hasSelectedRow: TableSelectors.getSelectedRows(state, ownProps).length > 0,
+    });
+
+    const mapDispatchToProps = (dispatch: IDispatch, ownProps: ITableHOCOwnProps) => ({
+        deselectRows: () => dispatch(TableHOCRowActions.deselectAll(ownProps.id)),
+    });
+
+    class TableWithActions extends React.Component<
+        ITableWithActionsProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>

```

Comment:
I would build a type with this `export type TableWithActionsType =  ITableWithActionsProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>>`

Reply:
I don't think we need to expose this type since it's all internal to the HOC and the class is never exposed